File: sdkwork-ai-sdk-typescript/src/api/embedding.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { EmbeddingRequest, EmbeddingResponse } from '../types';


export class EmbeddingApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create embeddings */
  async createEmbedding(body: EmbeddingRequest): Promise<EmbeddingResponse> {
    return this.client.post<EmbeddingResponse>(aiApiPath(`/embeddings`), body);
  }
}

export function createEmbeddingApi(client: HttpClient): EmbeddingApi {
  return new EmbeddingApi(client);
}

File: sdkwork-ai-sdk-go/api/embedding.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type EmbeddingApi struct {
    client *sdkhttp.Client
}

func NewEmbeddingApi(client *sdkhttp.Client) *EmbeddingApi {
    return &EmbeddingApi{client: client}
}

// Create embeddings
func (a *EmbeddingApi) CreateEmbedding(body sdktypes.EmbeddingRequest) (sdktypes.EmbeddingResponse, error) {
    raw, err := a.client.Post(AiApiPath("/embeddings"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.EmbeddingResponse
        return zero, err
    }
    return decodeResult[sdktypes.EmbeddingResponse](raw)
}
